wweval/R/combine_and_summarize_ww_data.R

```diff
@@ -0,0 +1,417 @@
+#'
+#' @description
+#' This function reports out summary statistics on the presence of and
+#' characteristics of wastewater data across all forecast
+#' dates and locations.
+#'
+#'
+#' @param ww_metadata a tibble containing one row for every forecast date
+#' location containing metrics averaged across that forecast date location
+#' about the wastewater data. This is the output of "get_add_ww_metadata".
+#' @param hosp_quantiles_filtered a large tibble containing all of the
+#' quantiled forecasts for the wastewater and hospital admissions only models
+#' after it has been filtered for convergence, wastewater data quality,
+#' manual exclusions, and dates that don't have any wastewater present
+#'
+#' @return a list containing a summary overall table, a summary by forecast
+#' date and a summary by state
+#' @export
+get_summary_ww_table <- function(ww_metadata, hosp_quantiles_filtered) {
+  # First, get the true number of forecast-date locations with wastewater
+  # in the current analysis
+  n_w_ww_actual <- hosp_quantiles_filtered |>
+    dplyr::filter(
+      .data$model_type == "ww"
+    ) |>
+    dplyr::distinct(.data$forecast_date, .data$location) |>
+    nrow()
+
+  n_no_ww_actual <- nrow(ww_metadata) - n_w_ww_actual
+
+  # Then get what would be expected
+
+  n_combinations <- nrow(ww_metadata)
+
+  n_combos_w_ww_data <- ww_metadata |>
+    dplyr::summarise(
+      n_ww_present = sum(.data$ww_data_present, na.rm = TRUE)
+    )
+
+  n_states_w_complete_ww_data <- ww_metadata |>
+    dplyr::group_by(.data$location) |>
+    dplyr::summarize(
+      complete_ww = all(.data$ww_data_present == 1)
+    ) |>
+    ungroup() |>
+    dplyr::summarize(n_complete_ww = sum(.data$complete_ww)) |>
+    dplyr::pull(.data$n_complete_ww)
+
+  n_states_w_no_ww_data <- ww_metadata |>
+    dplyr::group_by(.data$location) |>
+    dplyr::summarize(
+      no_ww = all(.data$ww_data_present == 0)
+    ) |>
+    dplyr::ungroup() |>
+    dplyr::summarize(n_zero_ww = sum(.data$no_ww)) |>
+    dplyr::pull(.data$n_zero_ww)
+
+  n_combos_w_hosp_conv_flags <- ww_metadata |>
+    dplyr::summarise(n_hosp_flags = sum(.data$any_flags_hosp, na.rm = TRUE)) |>
+    dplyr::pull(.data$n_hosp_flags)
+  n_combos_w_ww_conv_flags <- ww_metadata |>
+    dplyr::summarise(n_ww_flags = sum(.data$any_flags_ww, na.rm = TRUE)) |>
+    dplyr::pull(.data$n_ww_flags)
+
+  n_insuff_ww <- ww_metadata |>
+    dplyr::summarise(
+      n_ww_insuff = sum(.data$ww_sufficient == FALSE, na.rm = TRUE)
+    ) |>
+    dplyr::pull(n_ww_insuff)
+
+  n_ww_excluded <- ww_metadata |>
+    dplyr::summarise(
+      n_ww_excl = sum(.data$ww_exclude_manual, na.rm = TRUE)
+    ) |>
+    dplyr::pull(.data$n_ww_excl)
+
+  n_w_ww_expected <- ww_metadata |>
+    dplyr::mutate(
+      ww_expected = dplyr::case_when(
+        ww_data_present == 1 & is.na(.data$ww_exclude_manual) &
+          !isTRUE(.data$any_flags_hosp) & !isTRUE(.data$any_flags_ww) &
+          isTRUE(.data$ww_sufficient) ~ TRUE,
```

Comment:
In this situation, I think you probably want just plain `!` rather than `!isTRUE` as the former will be stricter.
```suggestion
          !(.data$any_flags_hosp) & !(.data$any_flags_ww) &
         isTRUE(.data$ww_sufficient) ~ TRUE,
```